src/software/amazon/ion/impl/SymbolTableReader.java

```diff
@@ -1353,4 +1354,13 @@ public void close() throws IOException
     {
         _current_state = S_EOF;
     }
+
+    public IntegerType getIntegerType()
+    {
+        if (stateType(_current_state) != IonType.INT)
+        {
+            return IntegerType.NONE;
+        }
+        return IntegerType.INT; // all of SymbolTable's integers are type int
```

Comment:
In practice, all of the integer fields (max ID, version, and symbol IDs)  that make up the in-memory symbol table implementations (e.g. SymbolTableImpl) are declared as `int`s in IonJava. The SymbolTableReader provides an implementation of the IonReader interface over a SymbolTable. The spec doesn't prescribe any size limitations for these fields, so there are existing problems within the implementation if it encounters a shared symbol table with more than `Integer.MAX_VALUE` versions or a local symbol table with more than `Integer.MAX_VALUE` symbols. If we think it's worth designing for that use case, I'd be happy to create an issue.